common/flatpak-installation.c

```diff
@@ -991,6 +991,75 @@ _ostree_collection_ref_free0 (OstreeCollectionRef *ref)
   ostree_collection_ref_free (ref);
 }
 
+static gboolean
+flatpak_installation_check_installed_ref_missing_related_ref (FlatpakInstallation *self,
+                                                              const gchar         *remote_name,
+                                                              gchar               *full_ref,
+                                                              GCancellable        *cancellable)
+{
+  g_autoptr (GPtrArray) installed_related_refs = NULL;
+  g_autoptr (GPtrArray) remote_related_refs = NULL;
+  g_autoptr (GError) local_error = NULL;
+  guint j, k;
+
+  installed_related_refs = flatpak_installation_list_installed_related_refs_sync (self,
+                                                                                  remote_name,
+                                                                                  full_ref,
+                                                                                  cancellable,
+                                                                                  &local_error);
+  if (local_error != NULL)
+    {
+      g_warning ("Unable to get installed related refs for %s: %s", full_ref, local_error->message);
+      return FALSE;
+    }
+
+  remote_related_refs = flatpak_installation_list_remote_related_refs_sync (self,
```

Comment:
Does this hit the network each time it's called? If so, that seems pretty sad, since this is run once per installed app, no?

Reply:
Not particularly. I looked at the call-map, and there is use of `xa.cache` metadata where-ever possible. However, I would still assume hit the network each time as the worst-case scenario though